eth/stagedsync/stage_headers.go

```diff
@@ -581,7 +596,20 @@ func verifyAndSaveNewPoSHeader(
 	}
 
 	currentHeadHash := rawdb.ReadHeadHeaderHash(tx)
-	if currentHeadHash == header.ParentHash {
+	if currentHeadHash == header.ParentHash || (header.ParentHash == cfg.hd.GetNextForkHash() && cfg.memoryOverlay) {
```

Comment:
Better to split it into two sequential `if` statements instead of using `||` and splitting it inside. Better readability that way